ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -693,6 +693,7 @@ private static void runTool(
                     tools.get(action).run(parameters, toolListener); // run tool
                 }
             } catch (Exception e) {
+                log.error("Failed to run tool {}", action, e);
```

Comment:
the next line, the error message will be return to the listener, is the log needed?

Reply:
Log is for debugging if later we need to deep dive to identify the root cause. And listener is for customer to be notified at that very moment.